packages/apps/plugins/plugin-map/src/components/MapDetectLocations.ts

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { CollectionType, TableType, type MapType } from '@braneframe/types';
+import { type DynamicSchema } from '@dxos/echo-schema';
+import { Filter, getSpace, useQuery } from '@dxos/react-client/echo';
+
+import { type Marker } from './MapControl';
+
+/**
+ * Find table rows that are in the same collection as the map.
+ * Return the ones with `latitude` and `longitude` fields,
+ * converted to `Marker`s.
+ *
+ * NOTE: This collects rows from all the collections the map is in,
+ * even if there's more than one. At the time of writing this behavior
+ * won't be exercised since being included in multiple collections
+ * is disabled in the UI.
+ *
+ * If we enable being in more than one collection again we should
+ * reconsider what behavior we actually want in that case.
+ */
+export const useMapDetectLocations = (map: MapType): Marker[] => {
+  const space = getSpace(map);
+
+  const parentCollections = useQuery<CollectionType>(
+    space,
+    [
+      Filter.schema(CollectionType),
+      (collection: any) => {
+        if (!(collection instanceof CollectionType)) {
+          return false;
+        }
+        return collection.objects.some((obj) => obj?.id === map.id);
+      },
+    ],
+    undefined,
+    [space],
+  );
+
+  const siblingIds = new Set(
+    parentCollections.flatMap((collection) => collection.objects.filter(hasId).map((obj) => obj.id)),
+  );
+
+  const siblingTables = useQuery<TableType>(
+    space,
+    [
+      Filter.schema(TableType),
+      (table: any) => siblingIds.has(table.id),
+      (table: any) => {
+        if (!(table instanceof TableType)) {
+          return false;
+        }
+        return table.schema ? schemaHasLatitudeAndLongitude(table.schema) : false;
+      },
+    ],
+    undefined,
+    [parentCollections],
+  );
+
+  const schemaIds: string[] = siblingTables.flatMap((table) => (table.schema ? [table.schema.id] : []));
+
+  const rows = useQuery(
+    // If there are no schema IDs that we're interesting in return nothing.
+    // (this is necessary since the or-based filter will always return true in this case):
+    schemaIds.length === 0 ? undefined : space,
+    Filter.or(...schemaIds.map((id) => Filter.typename(id))),
```

Comment:
Does filter have an idiomatic "reject all" option? If so I'll replace the above with:
```
  const rows = useQuery(
    space,
    schemas.length === 0 ? Filter.rejectAll : Filter.or(...schemaIds.map((id) => Filter.typename(id))),
```

Reply:
(Not looking for a literal `rejectAll`, just something with the same effect that's performant)